Language and style review (#17)
SUSE/BCI-docs: content/guides/container-suseconnect.md

# What is container-suseconnect?

[`container-suseconnect`](https://github.com/SUSE/container-suseconnect) is a
plugin available in all Base Container Images that ship with Zypper. When

> the plugin detects the host's SUSE Linux Enterprise Server, it automatically

Reviewer: ```suggestion
the plugin finds the host's SUSE Linux Enterprise Server registration credentials, it automatically
```

> registers the running container. This gives access the SUSE Linux

Reviewer: This is technically not correct. The container is not registered at all, because that would count towards the number of subscriptions that you are allowed to consume. `container-suseconnect` merely reuses your credentials and constructs the appropriate repositories from that.

Enterprise repositories. This includes additional modules and previous package
versions that are not part of the free SLE_BCI repository.

## How to use container-suseconnect

If you are running a registered SLES system with Docker, `container-suseconnect`
automatically detects and uses the subscription, without requiring any action on

> your part.

Reviewer: "on your part" sounds weird to me, but that might be just me.
```suggestion
your side.
```

Author: No, `part` is correct, see https://www.ldoceonline.com/dictionary/on-somebody-s-part-on-the-part-of-somebody

On openSUSE systems with Docker, you must copy the files `/etc/SUSEConnect` and
`/etc/zypp/credentials.d/SCCcredentials` from a registered SLES machine to your
local machine. Note that the `/etc/SUSEConnect` file is required only if you are
using RMT for managing your registration credentials.

> ## How to use container-suseconnect on non-SLE hosts or with Podman, Buildah or with nerdctl

Reviewer: Why not capitalize nerdctl as well?

Author: Because `netctl` is written in lowercase, see https://github.com/containerd/nerdctl

You need a registered SLES system to use `container-suseconnect` on non-SLE
hosts or with Podman, Buildah, or with nerdctl. This can be a physical machine,
a virtual machine, or the bci-base container with `SUSEConnect` installed and
registered. If you don't use RMT, copy `/etc/zypp/credentials.d/SCCcredentials`
to the development machine. Otherwise, copy both the
`/etc/zypp/credentials.d/SCCcredentials` and `/etc/SUSEConnect` files.

You can use the following command to obtain `SCCcredentials` (replace

> `REGISTRATION_CODE` with your SUSE Connect registration code):

Reviewer: I believe it is not referred to as the "SUSE Connect registration code", at least I never heard that term.
```suggestion
`REGISTRATION_CODE` with your SCC registration code)
```

> _REGISTRATION_CODE_ with your SUSE Connect registration code):

Reviewer: I would suggest to render it as verbatim, as is commonly done in markdown docs. Otherwise it will be displayed in italic, which imho reduces the recognition of that parameter.
```suggestion
`REGISTRATION_CODE` with your SUSE Connect registration code):
```

{{< tabs "obtain_scccredentials">}}
{{< tab "Docker">}}
```Shell
docker run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
```
{{< /tab >}}
{{< tab "Podman" >}}
```Shell
podman run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
```
{{< /tab >}}
{{< tab "nerdctl" >}}
```Shell
nerdctl run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
```
{{< /tab >}}
{{< /tabs >}}

If you are directly running a container based on a BCI, mount `SCCcredentials`
(and optionally `/etc/SUSEConnect`) in the correct destination. The following

> example show how to mount `SCCcredentials` in the current working directory:

Reviewer: ```suggestion
example shows how to mount `SCCcredentials` in the current working directory:
```

{{< /tab >}}
{{< /tabs >}}

Do not copy the `SCCcredentials` and `SUSEConnect` files into your container

> image t avoid inadvertently adding them to the final image. Use secrets instead,

Reviewer: ```suggestion
image to avoid inadvertently adding them to the final image. Use secrets instead,
```

as they are only available to a single layer and are not part of the built

> image. To do this, put copy of `SCCcredentials`  (and optionally `SUSEConnect`)

Reviewer: ```suggestion
image. To do this, put a copy of `SCCcredentials`  (and optionally `SUSEConnect`)
```
